### ISSUE TITLE
Error with comments in xml workspace save

### ISSUE THREAD

**Issue by lejoueur50** [google/blockly#6188]
**Describe the bug**
When a comment is added on a block, and contain `'` or `\n`, using `Blockly.Xml.domToText(Blockly.Xml.workspaceToDom(workspace))` will return a text that will throw an error when loaded

**To Reproduce**

1. Add a comment on any block, containing `'` or `\n`
2. Save workspace with `Blockly.Xml.domToText(Blockly.Xml.workspaceToDom(workspace))` functions
3. Load a new workspace with `Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(saved_xml), workspace);` and the generated xml text
4. Workspace don't appear, and an error is thrown

**Expected behavior**

The generated text should be correctly handled by `Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom())` functions, when the generated xml text is given to these functions, the workspace should load correctly ( with comments ) instead of throwing an error

**Screenshots**

![Capture](https://user-images.githubusercontent.com/56761209/172069146-cf633daf-a98a-456f-ad95-957df42760cb.JPG)

**Desktop :**
 - OS: Windows
 - Browser Mozilla
 - Version 101.0

**Stack Traces**

With `\n` in comment :
```
Uncaught SyntaxError: '' string literal contains an unescaped line break
```
With `''` in comment :
```
Uncaught SyntaxError: missing ) after argument list
```
With `'` in comment :
```
Uncaught SyntaxError: invalid regular expression flag b
```

**Additional context**

The generated xml string for the block in *screenshots* part is
```
<xml xmlns="https://developers.google.com/blockly/xml"><block type="event_message_sent" id="r:+Q_``KXq2Tl9O6v8Fc" x="90" y="90"><comment pinned="true" h="80" w="160">Hello 

World ''</comment></block></xml>
```

I save result of functions `Blockly.Xml.domToText(Blockly.Xml.workspaceToDom(workspace))` in a database, then restore workspace with `Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(saved_xml), workspace);`.

**Comment by lejoueur50** [comment]
Hello, thanks for the reply ;)

I made more tests to find the exact steps to reproduce the bug and found the problems :
First, when getting saved workspace from database and adding it to webpage, I used
```js
Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom('<%- saved_xml %>'), workspace);//Back-end
``` 
wich became, when sent to the client :
```js
Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom('<xml xmlns="https://developers.google.com/blockly/xml"><block type="event_message_sent" id="r:+Q_``KXq2Tl9O6v8Fc" x="90" y="90"><comment pinned="true" h="80" w="160">Hello 

World ''</comment></block></xml>'), workspace);//Front-end
```
Since `''` strings don't support multi-line, an error was thrown, so I corrected this problem with backticks ( `` ` `` ) strings.

The second problem was that Blockly use sometimes `` ` `` in blocks id in Xml, which conflict with backticks strings. The workaround was to call `` .replaceAll('`', '\''); `` and [remove blocks id from saves](https://developers.google.com/blockly/reference/js/Blockly.Xml#.workspaceToDom). `replaceAll()` function can also be used to fix incorrect users inputs in comments.

Seems like the bug doesn't come from Blockly, sorry for the trouble
Thank you for your help !
